### PR TITLE
Remove conda from R client

### PR DESCRIPTION
**Pull request by harupy**
Signed-off-by: harupy <hkawamura0130@gmail.com>

## Related Issues/PRs

#xxx

## What changes are proposed in this pull request?

Remove conda from the R client and update the documentation.

## How is this patch tested?

- [x] I have written tests (not required for typo or doc fix) and confirmed the proposed feature/bug-fix/change works.

Existing tests

## Does this PR change the documentation?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Click the `Details` link on the `Preview docs` check.
2. Find the changed pages / sections and make sure they render correctly.

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/pipelines`: Pipelines, Pipeline APIs, Pipeline configs, Pipeline Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language
- [x] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [x] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/R/mlflow/README.md [hunk 1]
```diff
@@ -10,19 +10,24 @@ Status](https://travis-ci.org/rstudio/mlflow.svg?branch=master)](https://travis-
...
   - Use MLflow to export models that can be served locally and remotely.
 
+## Prerequisites
+
+To use the MLflow R API, you must install [the MLflow Python package](https://pypi.org/project/mlflow/) and set the `MLFLOW_PYTHON_BIN` and `MLFLOW_BIN` environment variables as follows:
+
+```bash
+pip install mlflow
+export MLFLOW_PYTHON_BIN=$(which python)
+export MLFLOW_BIN=$(which python)
```
Comment (harupy) [inline mlflow/R/mlflow/README.md:20]: TODO: Automatically detect python and mlflow binaries.

File: .github/workflows/r.yml [hunk 2]
```diff
@@ -72,7 +72,10 @@ jobs:
         ./build-package.sh
     - name: Create test environment
       run: |
-        Rscript -e 'source(".create-test-env.R", echo=TRUE)'
+        pip install -e ../../..
```
Comment (dbczumar) [inline .github/workflows/r.yml:75]: Can we add a small comment explaining what's located at ../../..?

Comment (harupy) [reply .github/workflows/r.yml]: Replaced `../../..` with `git rev-parse --show-toplevel`.

File: docs/source/R-api.rst [hunk 3]
```diff
@@ -6,59 +6,27 @@ R API
...
-one-time setup step, you must run install_mlflow() to install these
-dependencies before calling other MLflow APIs.
+Optionally, you can set the ``MLFLOW_PYTHON_BIN`` and ``MLFLOW_BIN`` environment variables to specify the Python and MLflow binaries to use. By default, the R client automatically finds them using ``Sys.which('python')`` and ``Sys.which('mlflow')``.
 
-.. code:: r
+.. code-block:: bash
 
-   install_mlflow(python_version = "3.7")
+    export MLFLOW_PYTHON_BIN=/path/to/python/bin
+    export MLFLOW_BIN=/path/to/mlflow/bin
```
Comment (WeichenXu123) [inline docs/source/R-api.rst:23]: The example path seemingly should be:
```
    export MLFLOW_PYTHON_BIN=/path/to/python/bin/python
    export MLFLOW_BIN=/path/to/mlflow/bin/mlflow
```

File: docs/source/R-api.rst [hunk 4]
```diff
@@ -6,59 +6,27 @@ R API
...
-Alternatively, you can set MLFLOW_PYTHON_BIN and MLFLOW_BIN environment
-variables instead. MLFLOW_PYTHON_BIN should point to python executable
-and MLFLOW_BIN to mlflow cli executable. These variables allow you to
-use custom mlflow installation. Note that there may be some
-compatibility issues if the custom mlflow version does not match the
-version of the R package.
-
-Examples
---------
+You can use the R API to start the `user interface <mlflow_ui>`_, `create <mlflow_create_experiment>`_ and `search experiments <mlflow_search_experiments>`_, `save models <mlflow_save_model>`_, `run projects <mlflow_run_>`_ and `serve models <mlflow_rfunc_serve_>`_ among many other functions available in the R API.
```
Comment (WeichenXu123) [inline docs/source/R-api.rst:25]:
```suggestion
You can use the R API to start the `user interface <mlflow_ui>`_, `create experiment <mlflow_create_experiment>`_ and `search experiments <mlflow_search_experiments>`_, `save models <mlflow_save_model>`_, `run projects <mlflow_run_>`_ and `serve models <mlflow_rfunc_serve_>`_ among many other functions available in the R API.
```

File: mlflow/R/mlflow/README.Rmd [hunk 5]
```diff
@@ -20,32 +20,41 @@ knitr::opts_chunk$set(fig.path = "tools/readme/", dev = "png")
...
+pip install mlflow
+```
+
+Optionally, you can set the ``MLFLOW_PYTHON_BIN`` and ``MLFLOW_BIN`` environment variables to specify
+the Python and MLflow binaries to use. By default, the R client automatically finds them using
+``Sys.which("python")`` and ``Sys.which("mlflow")``.
+
+```{bash eval=FALSE}
+export MLFLOW_PYTHON_BIN=/path/to/python/bin
+export MLFLOW_BIN=/path/to/mlflow/bin
```
Comment (WeichenXu123) [inline mlflow/R/mlflow/README.Rmd:37]: ditto

File: mlflow/R/mlflow/README.md [hunk 6]
```diff
@@ -10,20 +10,32 @@ Status](https://travis-ci.org/rstudio/mlflow.svg?branch=master)](https://travis-
...
+pip install mlflow
+```
+
+Optionally, you can set the ``MLFLOW_PYTHON_BIN`` and ``MLFLOW_BIN`` environment variables to specify
+the Python and MLflow binaries to use. By default, the R client automatically finds them using
+``Sys.which("python")`` and ``Sys.which("mlflow")``.
+
+```bash
+export MLFLOW_PYTHON_BIN=/path/to/python/bin
+export MLFLOW_BIN=/path/to/mlflow/bin
```
Comment (WeichenXu123) [inline mlflow/R/mlflow/README.md:27]: ditto

File: docs/source/R-api.rst (hunk 3 above)
Comment (harupy) [reply docs/source/R-api.rst:23]: You're right. I'll fix.

Comment (harupy) [reply docs/source/R-api.rst]: fixed

File: docs/source/R-api.rst (hunk 4 above)
Comment (harupy) [reply docs/source/R-api.rst]: thanks for the catch! Fixed

**Review by dbczumar — APPROVED** [review]
@harupy LGTM! Can we try this out on Databricks MLR and confirm that it works before merging?

**Comment by harupy** [comment]
@dbczumar I tested on Databricks MLR. Databricks MLR patches `install_mlflow` when mlflow is imported via `library(mlflow)`. We need to skip this patch if mlflow version is >= 2.0. I filed a Jira ticket to track this issue.

**Review by WeichenXu123 — APPROVED** [review]
LGTM

**Pull request merged**